src/ColumnMetrics.js

```diff
@@ -99,7 +99,8 @@ function setDefferedColumnWidths(columns, unallocatedWidth, minColumnWidth) {
  * @param {number} width
  */
 function resizeColumn(metrics: ColumnMetricsType, index: number, width: number): ColumnMetricsType {
-  var column = metrics.columns[index];
```

Comment:
I dont think we should be doing this
inside columnMetrics, surely we already know if a column is an immutable or not?
Looks like the `ColumnMetricsType` definition needs to be update too - but assuming `metrics.columns: ImmutableList<Column>` and not `metrics.columns: ImmutableList<Column> | Aray<Column>`
obviously then means we need to trace back what is changing columns from ImmutableList -> Array?